File: Python/Tema_2/Boletin_3/Ejercicio05.py
def palindrome(cad):
    k = 0
    encontrado = False
    cad = cad.lower().strip()
    while not(encontrado) and k < len(cad) // 2:
        if cad[k] != cad[-k-1]: #accedemos a las ultimas posiciones de la cadena pero hay que restarle una porque empieza a contar desde -1 y no desde 0
            encontrado = True
        k += 1
        
    return not(encontrado)


print(palindrome('sal'))
print(palindrome('tenet'))
print(palindrome('radar'))
print(palindrome('reconocer'))
print(palindrome('ligar es ser agil'))

File: Java/Practicas/src/Boletin_03/Ejercicio_05.java
package Boletin_03;

public class Ejercicio_05 {
	public static void main(String[] args) {
		System.out.println(esPalindromo("ligar es ser agil"));
	}

	public static boolean esPalindromo(String cadena) {
		boolean encontrada = false;
		cadena = cadena.replace(" ", "");
		for (int n = 0; n < cadena.length() && !encontrada; n++) {
			if (cadena.charAt(n) != cadena.charAt(cadena.length() - n - 1)) {
				encontrada = true;
			}
		}
		return !encontrada;
	}

}
